engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1559,10 +1578,77 @@ private static boolean serializeAllTable(
             }
             logEntry.append(", usePrev=").append(usePrev).endl();
         }
+        return true;
+    }
 
+
+    /**
+     * Check if all the required column sources are {@link InMemoryColumnSource}.
+     */
+    private static boolean allColumnSourcesInMemory(
+            @NotNull final Table table,
+            @NotNull final String[] columnSources,
+            @Nullable final BitSet columnsToCheck) {
+        for (int colIdx = 0; colIdx < columnSources.length; colIdx++) {
+            if (columnsToCheck != null && !columnsToCheck.get(colIdx)) {
+                continue;
+            }
+            final ColumnSource<?> columnSource = table.getColumnSource(columnSources[colIdx]);
+            if (!(columnSource instanceof InMemoryColumnSource)) {
+                return false;
+            }
+        }
         return true;
     }
 
+    /**
+     * Serialize a single column of a table into a BarrageMessage.
+     * <p>
+     * This method is intended to be called from a thread pool executor.
+     */
+    private static void serializeColumn(
+            final String[] columnSources,
+            final int colIdx,
+            final boolean usePrev,
+            @NotNull final BarrageMessage snapshot,
+            @NotNull final Table table,
+            @NotNull final Object logIdentityObject,
+            @Nullable final BitSet columnsToSerialize) {
+        if (concurrentAttemptInconsistent()) {
```

Comment:
This is relying on a `ThreadLocal` at `StateImpl.threadState`. It will not be set correctly in your parallel jobs as is.

I think we need a mechanism to "export current attempt state" that we can them import in another tread. It should be done safely... we don't want job threads "ending" a snapshot.

Reply:
If there's no concurrent attempt active, we can simply ignore this stuff.
If there's one active, we need to capture and propagate the `activeConcurrentAttempt`, `concurrentSnapahostDepth=1`, and maybe `lastObservedClockValue` to the parallel worker thread. This should be cleared in the worker thread at the end of the task.

Reply:
Actually, let's just skip the fail-fast logic when working in parallel. If we assume that all the jobs run "almost immediately", the check is saving nothing. Keep it in the serial case... we can just pass a boolean param "serial" or something.